src/super_gradients/common/factories/activations_type_factory.py

```diff
@@ -0,0 +1,31 @@
+from typing import Union, Type, Mapping
+
+from super_gradients.common.factories.base_factory import AbstractFactory
+from super_gradients.training.utils.activations_utils import get_activation_type
+
+
+class ActivationsTypeFactory(AbstractFactory):
+    """
+    This is a special factory for torch.optim.Optimizer.
+    This factory does not instantiate an object but rather return the type, since optimizer instantiation
+    requires the model to be instantiated first
+    """
+
+    def get(self, conf: Union[str, Mapping]) -> Type:
+        """
+        Get a type.
+           :param conf: a configuration
+           if string - assumed to be a type name (not the real name, but a name defined in the Factory)
+           a dictionary is not supported, since the actual instantiation takes place elsewhere
+
+           If provided value is not one of the three above, the value will be returned as is
+        """
+        if isinstance(conf, str):
+            return get_activation_type(conf)
```

Comment:
The line of thinking for `ActivationsTypeFactory` was to benefit from `resolve_param`. 

A `get_activation_type()` method serves a few news:
* to being able to import existing activation functions from `torch.nn` and allowing some version-dependent dispatching (That's for future use. As an example: `if user asked for SiLU and torch version is < 1.6 import our own SiLU, otherwise use torch.nn.SiLU`).
* Handle lower-case activation functions names and well-known aliases (swish -> SiLU, leaky_relu, lrelu -> LeakyReLU)

We def. can merge them in a one class. It's just I'm not quite sure that implementation of `get_activation_type()` should be put inside the `ActivationsTypeFactory`. When used outside of configuration-based model instantiation it is easier to use `get_activation_type()`.